sklearn/base.py

```diff
@@ -273,7 +273,11 @@ def __repr__(self, N_CHAR_MAX=700):
     def __getstate__(self):
         try:
             state = super().__getstate__()
+            if state is None:
+                state = self.__dict__.copy()
         except AttributeError:
+            # TODO: Remove once Python < 3.11 is dropped, as there will never be
+            # an AttributeError
```

Comment:
this would still be raised for C extension types which don't inherit from PyObject though, right? One can have a C extension type w/o `__reduce__` for instance. It doesn't make much sense, but one can do that.

Reply:
Yes, right, I'll remove the comment.